### ISSUE TITLE
cmd/celestia: `dev` mode should be a flag on each node type (`full` | `light`) rather than a separate mode

### ISSUE THREAD

**Issue by renaynay** [celestiaorg/celestia-node#199]
We have a `dev` mode option in the CLI right now, but what it does is it basically runs an embedded dummy Tendermint node with a dummy app that generates blocks and feeds them to the celestia Full node. What we need is a `--dev` flag that can be called with the `full` node type. Keep in mind, the `light` node does not need a `dev` mode because it will work once connected to a bootstrap peer. 

For the `full --dev` mode, it can work the same as the current `dev` mode option, but just with the celestia app rather than the dummy kvstore that is currently used. Evan has a PR open (https://github.com/celestiaorg/celestia-node/pull/176) to replace the dummy kvstore app with celestia-app, so that can be merged / used for reference. 

For `light`, we need to be able to feed the `light` node a bootstrap peer that is already started (ideally another `full --dev` node) so that the `light` node can start syncing headers from the dummy full node. So basically, you should use the following command: 
`./celestia light init --bootstrap=<IP:PORT>` and then start the node. 

For example: 
```
./celestia light init --bootstrap=a5b96b333f698eddfb08db6507e801651738c9ca@0.0.0.0:26656 

./celestia light start
```
where `a5b96b333f698eddfb08db6507e801651738c9ca@0.0.0.0:26656` is the address of the full node. 

Basically, a dummy light node cannot run without being connected to something, but it doesn't really matter if the light node is connected to a bootstrap peer that is a dummy as long as the blocks are "valid".

## Questions: 

- [ ] Do we have a method to generate a dummy valid chain? If so, we need to integrate that with `full --dev` mode / celestia-app. This means we don't really need to have a `--dev` mode for `light` node as the headers will technically be valid.

cc @Wondertan

**Comment by renaynay** [comment]
Alternatively, `full --dev` mode can take a `--remote=<remote core node address>` in order to run the celestia-app. 

cc @evan-forbes

**Comment by renaynay** [comment]
Blockers to running a full / light node with celestia-app: 

- [ ] https://github.com/celestiaorg/celestia-node/pull/196 needs to be merged or this https://github.com/celestiaorg/celestia-node/pull/176
- [ ] we need `--bootstrap` flag that's able to take a bootstrap node and feed it to light node p2p config

Anything else? @Wondertan

**Comment by renaynay** [comment]
After talking with @Wondertan, we will close this issue as it is no longer necessary given new testing architecture plans.
